eslint.config.cjs

```diff
@@ -1,17 +1,29 @@
-module.exports = {
-  parser        : '@typescript-eslint/parser',
-  parserOptions : {
-    ecmaVersion : 'latest', // Allows the use of modern ECMAScript features
-    sourceType  : 'module', // Allows for the use of imports
+const tsParser = require("@typescript-eslint/parser");
+
+const tsPlugin = require("@typescript-eslint/eslint-plugin");
+const todoPlzPlugin = require('eslint-plugin-todo-plz');
+
+module.exports = [{
+  languageOptions: {
+    parser: tsParser,
+    parserOptions: {
+      ecmaFeatures: { modules: true },
+      ecmaVersion: 'latest',
+      project: './tsconfig.json',
+    },
   },
-  plugins: [
-    '@typescript-eslint',
-    'todo-plz' // for enforcing TODO formatting to require "github.com/TBD54566975/dwn-sdk-js/issues/"
-  ],
-  env: {
-    node    : true, // Enable Node.js global variables
-    browser : true
+  plugins: {
+    "@typescript-eslint": tsPlugin,
+    'todo-plz': todoPlzPlugin // for enforcing TODO formatting to require "github.com/TBD54566975/dwn-sdk-js/issues/"
   },
+  files: [
+    '**/*.ts'
+  ],
+  // IMPORTANT and confusing: `ignores` only exclude files from the `files` setting.
+  // To exclude *.js files entirely, you need to have a separate config object altogether. (See another `ignores` below.)
+  ignores: [
+    '**/*.d.ts',
```

Comment:
For my own knowledge, why are type definition files ignored?

Reply:
Hey @LiranCohen , those files are generated so we don't want to run the linter over them right?

Reply:
Ah ok yeah that makes sense!